runtime/gc_glue_java/StaccatoGCDelegate.hpp

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#if !defined(STACCATOGCDELEGATE_HPP_)
+#define STACCATOGCDELEGATE_HPP_
+
+#include "j9.h"
+#include "j9cfg.h"
+
+#if defined(J9VM_GC_REALTIME)
+
+#include "omr.h"
+#include "BaseNonVirtual.hpp"
+#include "EnvironmentRealtime.hpp"
+#include "RealtimeAccessBarrier.hpp"
+#include "RealtimeGCDelegate.hpp"
+
+class MM_StaccatoGC;
+
+class MM_StaccatoGCDelegate : public MM_BaseNonVirtual
+{
+private:
+	OMR_VM *_vm;
```

Comment:
Do we really need to have (OMR_VM *) in delegates? Looks like (J9JavaVM*) would be better choice. Did I missed something?